src/allocator/mod.rs

```diff
@@ -0,0 +1,48 @@
+// use crate::arch;
+use crate::*;
+
+// TODO: Find real value
+const PAGE_NUMBER: usize = 50;
+
+static mut PageUsage: [UsageFlags; PAGE_NUMBER] = [UsageFlags::Free; PAGE_NUMBER];
+
+#[repr(u8)]
+#[derive(Debug, PartialEq)]
+enum UsageFlags {
+    Used = 1,
+    Free = 0,
+}
+
+pub fn page_alloc() -> Option<usize> {
+    let mut index = 0;
+
+    // FIXME: Iterator
+    while index < 50 {
+        unsafe {
```

Comment:
Can we put the unsafe block just around the "pointer arithmetics"? Or does rustc disagree

Reply:
``PageUsage[index]`` is unsafe (static mut), line 26 is also unsafe. So I made a big unsafe block. But you're right it's better to see exactly where is the unsafe code in the most fine grained manner as possible. I will change this